File: examples/java/explicitDestination/Slave.java
/*
 * $Id: Slave.java 5059 2007-11-19 20:01:59Z mquinson $
 *
 * Copyright 2006,2007 Martin Quinson, Malek Cherier         
 * All rights reserved. 
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the license (GNU LGPL) which comes with this package. 
 */

import simgrid.msg.*;

public class Slave extends simgrid.msg.Process 
{
	public void main(String[] args) throws JniException, NativeException 
	{
		Msg.info("Hello !");
		
		BasicTask basicTask;
		Task receivedTask;
		
		while(true) 
		{ 
			receivedTask = Task.receive(Host.currentHost().getName());	
		
			if (receivedTask instanceof FinalizeTask) 
			{
				break;
			}
			
			basicTask = (BasicTask)receivedTask;
			Msg.info("Received \"" + basicTask.getName() + "\" ");
		
			Msg.info("Processing \"" + basicTask.getName() +  "\" ");	 
			basicTask.execute();
			Msg.info("\"" + basicTask.getName() + "\" done ");
		}
		
		Msg.info("Received Finalize. I'm done. See you!");
	}
}

File: examples/cxx/autoDestination/Slave.cxx
#include "Slave.hpp"
#include "FinalizeTask.hpp"
#include "BasicTask.hpp"

#include <Host.hpp>
#include <HostNotFoundException.hpp>

#include <Msg.hpp>

MSG_IMPLEMENT_DYNAMIC(Slave, Process);

int Slave::main(int argc, char** argv)
{
	info("Hello");
	
	Task* receivedTask;
	BasicTask* basicTask;
	
	while(true) 
	{ 
		receivedTask = Task::receive();	
	
		if(receivedTask->isInstanceOf("FinalizeTask")) 
		{
			delete receivedTask;
			break;
		}
	
		basicTask = reinterpret_cast<BasicTask*>(receivedTask);
	
		info(TEXT_("Received \"") + TEXT_(basicTask->getName()) + TEXT_("\" "));
	
		info(TEXT_("Processing \"") + TEXT_(basicTask->getName()) + TEXT_("\" "));
		
		basicTask->execute();
		
		info(TEXT_("\"") + TEXT_(basicTask->getName()) + TEXT_("\" done "));
	
		delete basicTask;
	}
		
	info("Received Finalize. I'm done. See you!");
	
	delete this;

	return 0;
}